src/schemas/json/hammerkit.json

```
@@ -1,239 +1,300 @@
-        },
-        "watch": {
-          "type": "boolean",
-          "description": "Watch the source files and restart the task on changes"
+          "$ref": "#/definitions/taskDeps"
         },
         "src": {
-          "$ref": "#/definitions/taskSource"
+          "$ref": "#/definitions/taskSrc"
         },
         "generates": {
-          "$ref": "#/definitions/taskGenerate"
+          "$ref": "#/definitions/taskGenerates"
         },
         "cmds": {
-          "description": "A list of commands in a task",
+          "title": "task commands",
+          "description": "Commands for the current task",
           "type": "array",
           "items": {
             "$ref": "#/definitions/taskCommand"
           }
         }
       }
     },
-    "dockerTask": {
+    "containerTask": {
+      "title": "container task",
+      "description": "Container task for the current build file\nhttps://no0dles.gitbook.io/hammerkit/task/container",
       "type": "object",
-      "required": ["cmds", "image"],
+      "required": ["image", "cmds"],
       "additionalProperties": false,
       "properties": {
         "description": {
-          "type": "string",
-          "description": "A description of the task"
-        },
-        "labels": {
-          "$ref": "#/definitions/labels"
+          "$ref": "#/definitions/taskCommand"
         },
         "extend": {
-          "type": "string",
-          "description": "Name of task that gets used a template"
+          "$ref": "#/definitions/taskExtend"
         },
-        "ports": {
-          "$ref": "#/definitions/taskPort"
+        "cache": {
+          "$ref": "#/definitions/taskCache"
+        },
+        "continuous": {
+          "$ref": "#/definitions/taskContinuous"
         },
         "envs": {
-          "$ref": "#/definitions/envMap"
+          "$ref": "#/definitions/taskEnvs"
         },
         "needs": {
-          "$ref": "#/definitions/taskNeed"
+          "$ref": "#/definitions/taskNeeds"
         },
         "deps": {
-          "$ref": "#/definitions/taskDependency"
+          "$ref": "#/definitions/taskDeps"
         },
-        "cache": {
-          "type": "string",
-          "description": "compare type for detecting changes in source files",
-          "enum": ["checksum", "modify-date", "none"]
+        "src": {
+          "$ref": "#/definitions/taskSrc"
         },
-        "continuous": {
-          "type": "boolean",
-          "description": "The task runs continuous and does not end without cancellation"
+        "generates": {
+          "$ref": "#/definitions/taskGenerates"
         },
-        "watch": {
-          "type": "boolean",
-          "description": "Watch the source files and restart the task on changes"
+        "cmds": {
+          "title": "task commands",
+          "description": "Commands for the current task",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/taskCommand"
+          }
         },
```

```suggestion
        "cmds": {
          "description": "Commands for the current task",
          "type": "array",
          "items": {
            "$ref": "#/definitions/taskCommand"
          }
        },
```

Mandatory fix: `title` is not required here, it's not an `object`.